providers/gce/gce_instances.go

```diff
@@ -192,7 +227,11 @@ func (g *Cloud) NodeAddresses(ctx context.Context, nodeName types.NodeName) ([]v
 		return nil, fmt.Errorf("error while querying for instance: %v", err)
 	}
 
-	return g.nodeAddressesFromInstance(instance)
+	addresses, err := g.nodeAddressesFromInstance(instance)
```

Comment:
`nodeAddressesFromInstance` is called from two places, the other one is in `func (g *Cloud) NodeAddressesByProviderID(ctx context.Context, providerID string) ([]v1.NodeAddress, error) {` and it seems you have to order it too

It is better that you just modify `nodeAddressesFromInstance` to return the addresses in the right order that make this change here

Reply:
Done